delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/resources/Resources.scala

```diff
@@ -423,40 +423,59 @@ object Resources {
     def update(u: UpdateResource) = {
       import u.jsonld._
       // format: off
-      for {
-        s                          <- stateWhereResourceIsEditable(u)
-        schemaRef                   = u.schemaOpt.getOrElse(ResourceRef.Latest(s.schema.iri))
-        (schemaRev, schemaProject) <- validate(u.id, expanded, schemaRef, s.project, u.caller)
-        time                       <- IOInstant.now
-      } yield ResourceUpdated(u.id, u.project, schemaRev, schemaProject, types, u.source, compacted, expanded, remoteContextRefs, s.rev + 1, time, u.subject, u.tag)
+      stateWhereResourceIsEditable(u).flatMap { s =>
+        val changesDetected = s.remoteContexts != remoteContextRefs || s.source != u.source
+        if(changesDetected) {
+          val schemaRef = u.schemaOpt.getOrElse(ResourceRef.Latest(s.schema.iri))
+          for {
+            (schemaRev, schemaProject) <- validate(u.id, expanded, schemaRef, s.project, u.caller)
+            time <- IOInstant.now
+          } yield ResourceUpdated(u.id, u.project, schemaRev, schemaProject, types, u.source, compacted, expanded, remoteContextRefs, s.rev + 1, time, u.subject, u.tag)
+        } else {
+          // If no changes have been detected on the resource itself, we fallback on other commands or raise an error
+          (u.schemaOpt, u.tag) match {
+            case (Some(schema), tagOpt) => updateResourceSchema(UpdateResourceSchema(u.id, u.project, schema, u.rev, u.caller, tagOpt))
+            case (None, Some(tagValue)) => tag(TagResource(u.id, u.project, u.schemaOpt, s.rev, tagValue, u.rev, u.subject))
+            case (None, None) => IO.raiseError(NoChangeDetected(s))
```

Comment:
It would be better to change the signature of the evaluate method instead of raising an error we recover afterwards.
I made some tests about it but those are some impacting changes and it is better to postpone them waiting for the other big changes to be merged